src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepository.kt

```diff
@@ -27,11 +27,23 @@ import javax.inject.Inject
 @Suppress("UnstableApiUsage")
 open class NexusRepository @Inject constructor(@Input val name: String, project: Project) {
 
+    @Internal
+    val s01 = NexusHost.S01
+
     @Input
-    val nexusUrl = project.objects.property<URI>()
+    val nexusHost = project.objects.property<NexusHost>().apply {
+        set(NexusHost.OSS)
+    }
 
     @Input
-    val snapshotRepositoryUrl = project.objects.property<URI>()
+    val nexusUrl = project.objects.property<URI>().apply {
+        set(nexusHost.map { it.resolve("/service/local/") })
+    }
+
+    @Input
+    val snapshotRepositoryUrl = project.objects.property<URI>().apply {
+        set(nexusHost.map { it.resolve("/content/repositories/snapshots/") })
+    }
```

Comment:
I think having three properties is weird because it would allow configuring a host that's not even used in `nexusUrl` and `snapshotRepositoryUrl`.

Reply:
`nexusUrl` would be the default only. It wouldn't apply if you override any of the URLs directly.